packages/headless-components/cms/src/react/CmsCollection.tsx

```diff
@@ -70,53 +75,252 @@ export const Root = React.forwardRef<HTMLDivElement, RootProps>(
   },
 );
 
+
 /**
- * Props for CmsCollection.Items headless component
+ * Props for CmsCollection.Items component
  */
 export interface ItemsProps {
-  /** Render prop function that receives collection items data */
-  children: (props: ItemsRenderProps) => React.ReactNode;
+  children: React.ReactNode;
+  emptyState?: React.ReactNode;
+  asChild?: boolean;
+  className?: string;
+  // TODO: add infiniteScroll and pageSize when we have pagination
+}
+
+/**
+ * Main container for the collection items display with support for empty states.
+ *
+ * @component
+ * @example
+ * ```tsx
+ * // Basic usage with empty state
+ * <CmsCollection.Items
+ *   emptyState={<div>No items found</div>}
+ *   className="space-y-4"
+ * >
+ *   <CmsCollection.ItemRepeater>
+ *     <CmsItem.Field fieldId="title" className="font-semibold mb-2" />
+ *     <CmsItem.Field fieldId="description" className="text-gray-600" />
+ *   </CmsCollection.ItemRepeater>
+ * </CmsCollection.Items>
+ *
+ * // Using asChild pattern
+ * <CmsCollection.Items asChild emptyState={<div>No data available</div>}>
+ *   <section className="grid grid-cols-3 gap-4">
+ *     <CmsCollection.ItemRepeater>
+ *       <CmsItem.Field fieldId="title" />
+ *       <CmsItem.Field fieldId="description" />
+ *     </CmsCollection.ItemRepeater>
+ *   </section>
+ * </CmsCollection.Items>
+ * ```
+ */
+export const Items = React.forwardRef<HTMLElement, ItemsProps>((props, ref) => {
+  const { children, emptyState, asChild, className, ...otherProps } = props;
+
+  return (
+    <CoreCmsCollection.Items>
+      {({ items, isLoading, error }) => {
+        // Don't render anything during loading or error states
+        if (isLoading || error) {
+          return null;
+        }
+
+        // Show empty state when no items
+        if (items.length === 0) {
+          return emptyState || null;
+        }
+
+        const dataAttributes = {
+          'data-testid': TestIds.cmsCollectionItems,
+          'data-empty': items.length === 0,
+        };
+
+        if (asChild) {
+          return (
```

Comment:
The AsChildSlot component simplifies handling of the `asChild` prop.
If `asChild` is false, it renders its children directly.
If `asChild` is true, it renders the provided custom element instead.

You can just change it to

```
        return (
            <AsChildSlot
              ref={ref}
              asChild={asChild}
              className={className}
              customElement={children}
              {...dataAttributes}
              {...otherProps}
            >
              <div>{children}</div>
            </AsChildSlot>
        );
```